compiler/rustc_lexer/src/lib.rs

```diff
@@ -414,9 +414,7 @@ impl Cursor<'_> {
 
             // Numeric literal.
             c @ '0'..='9' => {
-                let literal_kind = self.number(c);
-                let suffix_start = self.pos_within_token();
-                self.eat_literal_suffix();
```

Comment:
Looks like `eat_literal_suffix` is still always called in this position, so no need to move it into `fn number`.